docs/tools/config_system.md

```diff
@@ -34,28 +146,40 @@ For example:
 }
 ```
 
-The JSON fragment above defines three configuration parameters named `param1`, `param2` and `param3`. There are two ways to define a configuration parameter:
+You define a configuration parameter by specifying it's name as the key and specifying it's value either with a description object or by value. The JSON fragment above defines three configuration parameters named `param1`, `param2` and `param3`.
 
-- *The short way:* by name and value. `param3` above is an example of a short definition for a parameter named `param3` with value `10`.
-- *The long way:* by name and description (another JSON object), like `param1` and `param2` above. The JSON description object can have the following keys:
-    - `help`: an optional help message that describes the purpose of the parameter.
-    - `value`: an optional field that defines the value of the parameter.
-    - `required`: an optional key that specifies whether the parameter **must** be given a value before compiling the code (`false` by default). It's not possible to compile a source tree with one or more required parameters that don't have a value. Generally, it makes sense to define a required parameter only when it doesn't have a `value` key.
-    - `macro_name`: an optional name for the macro defined at compile time for this configuration parameter. The configuration system will automatically figure out the corresponding macro name for a configuration parameter, but the user can override this automatically computed name by specifying `macro_name`.
+Above, the configuration parameters `param1` and `param2` are defined using a description object. The following keys are supported on the description object:
+  - `help`: an optional help message that describes the purpose of the parameter.
+  - `value`: an optional field that defines the value of the parameter.
```

Comment:
But you said above that "A macro definition will not be generated for a parameter that doesn't have a value." If that's true, how can the value be optional? Or doesn't it make the parameter useless without a value?

Reply:
Related: https://github.com/ARMmbed/Handbook/pull/463/files#r176273626

Reply:
Yeah, I removed that note. Does that help?